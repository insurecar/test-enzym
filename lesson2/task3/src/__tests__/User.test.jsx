import React from "react";
import { shallow } from "enzyme";
import User from "../User";
import { getUserData, userGateway } from "../usersGateway";
import "jsdom-global/register";

jest.mock("../usersGateway", () => {
  return {
    getUserData: jest.fn(() => Promise.resolve()),
  };
});

describe("User", () => {
  it("should not display", () => {
    const wrappedComponent = shallow(<User />);

    expect(wrappedComponent.find(".user").exists()).toEqual(false);
  });

  it("should  display", () => {
    const wrappedComponent = shallow(<User />);

    wrappedComponent.setState({
      userData: {
        avatar_url: "ndjfbg",
        name: "Ivan",
        location: "Lviv",
      },
    });

    expect(wrappedComponent.find(".user").exists()).toBeTruthy();
    expect(wrappedComponent.find(".user")).toMatchSnapshot();
  });

  // it("should  snapshot", () => {
  //   const wrappedComponent = shallow(<User />);

  //   wrappedComponent.setState({
  //     userData: {
  //       avatar_url: "ndjfbg",
  //       name: "Ivan",
  //       location: "Lviv",
  //     },
  //   });

  //   expect(wrappedComponent.find(".user")).toMatchSnapshot();
  // });

  it("should  get function", () => {
    // shallow(<User />);

    expect(getUserData).toBeCalled();
  });
});
