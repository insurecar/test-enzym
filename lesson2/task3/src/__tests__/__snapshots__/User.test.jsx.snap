// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should  display 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
    src="ndjfbg"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      Ivan
    </span>
    <span
      className="user__location"
    >
      Lviv
    </span>
  </div>
</div>
`;

exports[`User should  get function 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
    src="ndjfbg"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      Ivan
    </span>
    <span
      className="user__location"
    >
      Lviv
    </span>
  </div>
</div>
`;

exports[`User should  snapshot 1`] = `
<div
  className="user"
>
  <img
    alt="User Avatar"
    className="user__avatar"
    src="ndjfbg"
  />
  <div
    className="user__info"
  >
    <span
      className="user__name"
    >
      Ivan
    </span>
    <span
      className="user__location"
    >
      Lviv
    </span>
  </div>
</div>
`;
